tests/functional/test_clidriver.py

```diff
@@ -103,6 +103,21 @@ def test_imds_region_is_used_as_fallback_with_v2_support(self):
         self.driver.main(['ec2', 'describe-instances'])
         self.assertEqual(capture.region, 'us-mars-2')
 
```

Comment:
I don't think it was included in the parent commit, but we should have unit test for the machine platform and make sure `source` gets included in the user agent extra. We can also add that in another PR, but something I noticed.

Reply:
I'll add `source` check in the PR where I'll actually read it form file